lib/services/arweave/arweave_service.dart

```diff
@@ -130,17 +132,25 @@ class ArweaveService {
       try {
         final entityType = transaction.getTag(EntityTag.entityType);
 
+        if (responses[i].statusCode != 200) {
+          throw Exception(
+            'Failed to fetch entity data for tx ${entityTxs[i].id}',
```

Comment:
```suggestion
            'Failed to fetch entity data for tx ${transcation.id}',
```
As we already have `entityTxs[i]` defined before I think that make sense to use it.

Reply:
Similar suggestion can be applied to `responses[i]` that we use 2 times, it can increase readability, but it's up to you. I think it's readable enough the way it is.